esp/services/ws_packageprocess/ws_packageprocessService.cpp

```diff
@@ -681,11 +749,15 @@ bool CWsPackageProcessEx::readPackageMapString(const char *packageMapString, Str
     return true;
 }
 
-void CWsPackageProcessEx::getPkgInfoById(const char *packageMapId, IPropertyTree* tree)
+void CWsPackageProcessEx::getPkgInfoById(const char *target, const char *packageMapId, IPropertyTree* tree)
 {
     if (!packageMapId || !*packageMapId)
         return;
 
+    StringBuffer scopedPMID;
+    if (target && *target)
+        packageMapId = scopedPMID.append(target).append("::").append(packageMapId).str();
```

Comment:
setf("%s::%s, target, packageMapId) assuming packageMapId is a char * 
2 less StringBuffer method calls and (this might be subjective) easier to read.